src/state/selectors/selectors.ts

```diff
@@ -210,7 +211,20 @@ export const combineTooltipPayload = (
     const finalNameKey: DataKey<any> | undefined = settings?.nameKey; // ?? tooltipAxis?.nameKey;
     let tooltipPayload: unknown;
     if (tooltipAxis?.dataKey && !tooltipAxis?.allowDuplicatedCategory && Array.isArray(sliced)) {
-      tooltipPayload = findEntryInArray(sliced, tooltipAxis.dataKey, activeLabel);
+      if (activeLabel || !sliced?.length) {
```

Comment:
why these conditions?

Reply:
From my tests and debugging:
<img width="825" alt="image" src="https://github.com/user-attachments/assets/841a6035-053c-410f-94cc-c7a052fbd1fe" />

when the points are on the same X, the `activeLabel` is null. Sometimes if I hover over the last element(100,300) the active label gets populated, but it's a very rare event.

I assume this might indicate the source of the issue.